metalearners/xlearner.py

```diff
@@ -99,31 +99,36 @@ def fit_all_nuisance(
 
         qualified_fit_params = self._qualified_fit_params(fit_params)
 
-        self._cvs: list = []
+        if not synchronize_cross_fitting:
+            raise ValueError()
+
+        self._cv_split_indices = self._split(X)
+        self._treatment_cv_split_indices = {}
 
         for treatment_variant in range(self.n_variants):
             self._treatment_variants_indices.append(w == treatment_variant)
-            if synchronize_cross_fitting:
-                cv_split_indices = self._split(
-                    index_matrix(X, self._treatment_variants_indices[treatment_variant])
+            treatment_indices = np.where(
```

Comment:
This is an opaque way of turning an array `[True, True, False, False, True]` to an array `[0, 1, 4]`. Not sure if there's a neater way of doing that.

Reply:
`[index for index, value in enumerate(vector) if value]`  would work too, I guess, but I like the np.where :)